src/components/QuestionAnswer/QuestionAnswer.jsx

```diff
@@ -73,21 +73,36 @@ export const QuestionAnswer = () => {
             })}
         </div>
         <div className="nxt-btn-container">
-          {currentQuestion < 4 ? (
+          {currentQuestion < questions.length - 1 ? (
             <button
               className="nxt-qstn button btn-primary cursor"
               onClick={() =>
                 quizDispatch({
-                  type: "NEXT_QUESTION"
+                  type: "NEXT_QUESTION",
+                  payload: {
+                    qns: questions[currentQuestion]?.question,
+                    ans: selectedAnswer
+                  }
                 })
               }
             >
               Next Question
             </button>
           ) : (
             <Link to="/result">
-              <button className="nxt-qstn button btn-primary cursor">
-                Next Question
+              <button
+                className="nxt-qstn button btn-primary cursor"
+                onClick={() =>
+                  quizDispatch({
+                    type: "RESULT_PAGE",
+                    payload: {
+                      qns: questions[currentQuestion]?.question,
```

Comment:
just a suggestion- I believe you can come up with better classnames also for the same term using too many different names might confuse you and the reader as well.
[ I noticed you have used used "qsn" "qstn"  "qns" for denoting question at different places]